src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/DelegateTests.cs

```diff
@@ -195,5 +195,233 @@ public static void InvokeFunctionAcceptingArrayTypes(Function objectPrototype, s
+                callback: null,
+                eventFactory:function(data){
+                    return {
+                        data:data
+                    };
+                },
+                fireEvent: function (evt) {
+                    this.callback(evt);
+                }
+            };");
+            var dispatcher = (JSObject)factory.Call();
+            var temp = new bool[100];
+            Action<JSObject> cb = (JSObject envt) =>
+            {
+                var data = (int)envt.GetObjectProperty("data");
+                temp[data] = true;
+            };
+            dispatcher.SetObjectProperty("callback", cb);
+            var evnt0 = dispatcher.Invoke("eventFactory", 0);
+            var evnt1 = dispatcher.Invoke("eventFactory", 1);
+            dispatcher.Invoke("fireEvent", evnt0);
+            dispatcher.Invoke("fireEvent", evnt0);
+            dispatcher.Invoke("fireEvent", evnt1);
+            Assert.True(temp[0]);
+            Assert.True(temp[1]);
+
+            for (int i = 0; i < 100; i++)
+            {
+                var evnti = dispatcher.Invoke("eventFactory", i);
+                dispatcher.Invoke("fireEvent", evnti);
+                Runtime.InvokeJS("gc();");// needs v8 flag --expose-gc
+            }
+        }
+
+        [Fact]
+        public static void NullDelegate()
+        {
+            var tcs = new TaskCompletionSource<int>();
+            var factory = new Function("delegate", "callback", @"
+                callback(delegate);
+            ");
+
+            Delegate check = null;
+            Action<Delegate> callback = (Delegate data) =>
+            {
+                check = data;
+            };
+            Assert.Null(check);
+            factory.Call(null, null, callback);
+            Assert.Null(check);
+        }
+
+        [Fact]
+        public static async Task ResolveStringPromise()
+        {
+            var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    resolve('foo');
+                  }, 300);
+                });");
+
+            var promise = (Task<object>)factory.Call();
+            var value = await promise;
+
+            Assert.Equal("foo", (string)value);
+            
+        }
+
+        [Fact]
+        public static async Task ResolveJSObjectPromise()
+        {
+            for (int i = 0; i < 10; i++)
+            {
+                var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    resolve({foo:'bar'});
+                  }, 300);
+                });");
+
+                var promise = (Task<object>)factory.Call();
+                var value = (JSObject)await promise;
+
+                Assert.Equal("bar", value.GetObjectProperty("foo"));
+
+                Runtime.InvokeJS("gc();");// needs v8 flag --expose-gc
```

Comment:
Is 10 runs enough to actually expose whatever gc bug was happening here?

Reply:
It was bug with jsHandle math which this helped me to test. 
I added GC just to try that Function would get collected.